src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -699,26 +697,7 @@ private TestMethodInfo ResolveTestMethodInfoForDiscovery(TestMethod testMethod,
         MethodInfo methodInfo = GetMethodInfoForTestMethod(testMethod, testClassInfo);
 
         // Let's build a fake options type as it won't be used.
```

Comment:
@Evangelink Here, things like timeout, test method attribute were not calculated. With the refactoring of the logic into TestMethodInfo, they will be calculated.

Is there a perf concern here for discovery? Maybe we can use `Lazy<T>`?